src/IO/Observer/Actions/ObserverRegistration.hpp

```diff
@@ -136,6 +189,68 @@ struct RegisterReductionNodeWithWritingNode {
   }
 };
 
+struct DeregisterReductionNodeWithWritingNode {
+  template <typename ParallelComponent, typename DbTagsList,
+            typename Metavariables, typename ArrayIndex>
+  static void apply(db::DataBox<DbTagsList>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const observers::ObservationKey& observation_key,
+                    const size_t caller_node_id) noexcept {
+    if constexpr (tmpl::list_contains_v<
+                      DbTagsList, Tags::NodesExpectedToContributeReductions>) {
+      auto& my_proxy =
+          Parallel::get_parallel_component<ParallelComponent>(cache);
+      const auto node_id =
+          static_cast<size_t>(Parallel::my_node(*my_proxy.ckLocalBranch()));
+      ASSERT(node_id == 0,
+             "Only node zero, not node "
+                 << node_id
+                 << " should deregister other nodes in the reduction");
+
+      db::mutate<Tags::NodesExpectedToContributeReductions>(
+          make_not_null(&box),
+          [&caller_node_id, &observation_key](
+              const gsl::not_null<
+                  std::unordered_map<ObservationKey, std::set<size_t>>*>
+                  reduction_observers_registered_nodes) noexcept {
+            if (UNLIKELY(reduction_observers_registered_nodes->find(
+                             observation_key) ==
+                         reduction_observers_registered_nodes->end())) {
+              ERROR(
+                  "Trying to deregister a node associated with an unregistered "
+                  "observation key: "
+                  << observation_key);
+            }
+            if (UNLIKELY(
+                    reduction_observers_registered_nodes->at(observation_key)
```

Comment:
[optional] You use this value a lot.  Maybe look it up once and store a reference or an iterator.